elastic/kibana
# [Response Ops][Alerting] Deleting ad hoc run task if the last schedule entry ends in a timeout

## Summary

Fixing bug where ad hoc run task was not getting deleted if running the last schedule entry and the run times out.

## To Verify
1. Create a detection rule
2. Add a delay in the ad hoc task runner
```
--- a/x-pack/plugins/alerting/server/task_runner/ad_hoc_task_runner.ts
+++ b/x-pack/plugins/alerting/server/task_runner/ad_hoc_task_runner.ts
@@ -530,6 +530,7 @@ export class AdHocTaskRunner {
     } catch (err) {
       runMetrics = asErr(err);
     }
+    await new Promise((resolve) => setTimeout(resolve, 3100000));
     await this.processAdHocRunResults(runMetrics);
```
3. Schedule a backfill for the rule with only one schedule entry

```
POST https://localhost:5601/internal/alerting/rules/backfill/_schedule
[
  {
    "rule_id": <ruleId>,
    "start": "2024-07-03T13:05:00.000Z"
  }
]
```

4. Wait for the run to get cancelled then verify that the task was deleted.
